### ISSUE TITLE
linter: false positive of "no-unused-expressions" for arrow functions with implied returns

### ISSUE THREAD

**Issue by djytw** [oxc-project/oxc#7584]
### What version of Oxlint are you using?

0.14.0

### What command did you run?

npx oxlint test.ts -D no-unused-expressions

### What does your `.oxlint.json` config file look like?

no config file

### What happened?

test.ts:

```typescript
const _func = (value: number) => value + 1;
```
```
  × typescript-eslint(no-unused-expressions): Disallow unused expressions
   ╭─[test.ts:1:34]
 1 │ const _func = (value: number) => value + 1;
   ·                                  ─────────
   ╰────
  help: Consider removing this expression

Finished in 3ms on 1 file with 98 rules using 20 threads.
Found 0 warnings and 1 error.
```
